lib/view_component/slotable.rb

```diff
@@ -356,9 +356,11 @@ def set_slot(slot_name, slot_definition = nil, *args, &block)
 
       # If class
       if slot_definition[:renderable]
+        args = add_slot_iterator_options(args, slot_definition[:renderable], slot_name) if args.any?
         slot.__vc_component_instance = slot_definition[:renderable].new(*args)
       # If class name as a string
       elsif slot_definition[:renderable_class_name]
+        args = add_slot_iterator_options(args, self.class.const_get(slot_definition[:renderable_class_name]), slot_name) if args.any?
```

Comment:
What if instead of having `if args.any?` at the call sites we had a guard at the top of the `add_slot_iterator_options` method that returned args if args is empty?

Reply:
@corasaurus-hex I have moved the guard clause to the method